servers/rendering/renderer_rd/shaders/scene_forward_lights_inc.glsl

```diff
@@ -253,7 +253,7 @@ float sample_directional_pcf_shadow(texture2D shadow, vec2 shadow_pixel_size, ve
 
 	mat2 disk_rotation;
 	{
-		float r = quick_hash(gl_FragCoord.xy) * 2.0 * M_PI;
+		float r = quick_hash(gl_FragCoord.xy + frame * 3) * 2.0 * M_PI;
```

Comment:
No rendering books were harmed in the making of this dither jittering :slightly_smiling_face: 

On a more serious note, this is the same algorithm I used in my initial proof of concept (before TAA was merged). This algorithm works pretty well when TAA is enabled, but is not always ideal when TAA is disabled (at least at lower framerates). Any ideas for cheap jittering patterns that could provide better results.

Reply:
The way the current per-pixel rotation works is actually already optimized to work with TAA. What you want to do is instead increase r (the rotation amount) by the proportion of the TAA cycle you are in e.g. TAA_FRAME/NUMBER_OF_JITTERS. So if you use 16 different jittered samples, you would use ``(frame %16)/16``. Then, add it to the result of quick_hash before multiplying by TAU. 

Remember ``r`` represents a rotation of the Vogel disk. It is not a dither. We rotate the Vogel disk randomly per pixel to break up patterns, but we also want to rotate it over time. 

The beauty of the Vogel disk is that, when rotated, the path created by each sample will not overlap the path from another. So by using a rotation, you are able to spread the samples out and maximize the benefit of the TAA

Reply:
Done. I pushed a second commit so it can be reviewed separately.

Visual comparisons: https://imgsli.com/MTExNzA1, https://imgsli.com/MTExNzA2

Reply:
> Even with TAA enabled, when jittering is enabled, I can sometimes see diagonal (but mostly vertical) lines appearing on the dithering pattern sometimes. I don't know why.

I figured it out in my review below.